### PR TITLE
[TH2-366] Correct the README file

### PR DESCRIPTION
**Pull request by OptimumOpium**
The information about codec configuration, implementations loading and pins definition was corrected

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,42 +1,106 @@
...
 
 This setting can be overridden in a custom config for the application using `decodeProcessorType` parameter.
 
-## Requried pins
+## Bootstrap parameters
 
-Every type of connections is presented two subscribe and publish pins. 
-Configuration should include at least one at a time pins for every types (minimal number of them is 8)
+These parameters specify the codec to be used for the messages decoding/encoding and the mode which should be used.
+Their should be defined in the `custom-config` section of the component configuration.
```
Comment (jean-michel-perli-sonsin) [inline README.md:59]: **They** should be defined in the `custom-config` section of the component configuration.

File: README.md [hunk 2]
```diff
@@ -1,42 +1,106 @@
...
 
+```yaml
+codecClassName: fully.qualified.class.name.for.Factory
+decodeProcessorType: CUMULATIVE
+```
+
+## Codec implementation parameters
+
+These parameters will be passed to the actual codec implementation to configure its behavior.
+A codec might not have any parameters to configure. In that case, you can omit adding those parameters.
```
Comment (jean-michel-perli-sonsin) [inline README.md:69]: **It's possible** that a codec might not have any parameters to configure. In **this** case, you can omit adding those parameters.

File: README.md [hunk 3]
```diff
@@ -1,42 +1,106 @@
...
+
+The codec implementation parameters should be located in the container's `/home` directory and stored in the file named `codec_config.yml`.
+It has simple key-value format in YAML.
+```yaml
+---
+param1: value1
+param2: value2
+```
+The set of parameters depends on the codec implementation is used.
```
Comment (jean-michel-perli-sonsin) [inline README.md:78]: The set of parameters depends on the codec implementation **that** is used.

File: README.md [hunk 4]
```diff
@@ -1,42 +1,106 @@
...
+
+## Required pins
+
+Every type of connection has two `subscribe` and `publish` pins.
+The first one is used to receive messages to decode/encode. The second one - to send decoded/encoded messages further.
```
Comment (jean-michel-perli-sonsin) [inline README.md:83]: The first one is used to receive messages to decode/encode **while the** second one **is used** to send decoded/encoded messages further.

File: README.md [hunk 5]
```diff
@@ -76,8 +140,8 @@ Let's consider some examples of routing in codec box.
 ### Split on 'publish' pins
 
 For example, you have got big source data stream, and you want to split them on some pins via session alias.
```
Comment (jean-michel-perli-sonsin) [inline README.md:142]: For example, you **got** a big source data stream, and you want to split them **into** some pins via session alias.

File: README.md [hunk 6]
```diff
@@ -103,4 +167,6 @@ spec:
             - field-name: session_alias
               expected-value: second_session_alias
               operation: EQUAL
-```
\ No newline at end of file
+```
+
+The filtering can also be performed for pins with  `subscribe` attribute.
```
Comment (jean-michel-perli-sonsin) [inline README.md:172]: The filtering can also be **applied** for pins with `subscribe` attribute.

**Review by jean-michel-perli-sonsin — CHANGES_REQUESTED** [review]

File: README.md [hunk 7]
```diff
@@ -1,42 +1,106 @@
 # How it works
 
-The th2 Codec component is responsible for encoding and decoding the messages. It is operating two instances of encoder/decoder pairs, one for operational purposes and one for general conversion.
+The th2 Codec component is responsible for encoding and decoding the messages.
+It operates two instances of encoder/decoder pairs, one for operational purposes and one for general conversion.
 
-Encoding and decoding is performed according to the scheme "one input queue and one or more output queues". Output queues can have filters, depending on which the output of the encoder and decoder can be partially or completely filtered out. The metadata of the message and its fields are used as filter parameters.
+Encoding and decoding is performed according to the scheme "one or more input pins and one or more output pins".
+Both types of pins can have filters, depending on which the input / output of the encoder / decoder can be partially or completely filtered out.
```
[the remark this answers is not in the document]
Comment (OptimumOpium) [reply README.md:7]: I would try to say that the input and output for decoders and encoders might be filtered depending on the filters each pin has.

Will the following option be correct?
> Both types of pins may have filters. The input / output of the encoder and decoder can be partially or entirely filtered out depending on filters the pin has.

[the remark this answers is not in the document]
Comment (jean-michel-perli-sonsin) [reply README.md:7]: The following option is correct and does not change the semantic. I would just add a "which" before the word filters (depending on which filters...), the rest of the sentence is right.

**Review by jean-michel-perli-sonsin — CHANGES_REQUESTED** [review]
Please refer to the previous comments that I made after the topic "running", the other points corrected are ok now.

File: README.md [hunk 8]
```diff
@@ -1,42 +1,106 @@
...
+
+with the content equals to the fully-qualified class name of your factory implementation.
+
+_If you have several implementations of that interface their fully-qualified names should be written in that file each one on the new line._
+
 
 # Configuration
 
-Codec has got four types of connection: stream and general for encode and decode functions.
+Codec has four types of connection: stream and general for encode and decode functions.
```
[the remark this answers is not in the document]
Comment (OptimumOpium) [reply README.md:43]: Why `have`? `Codec` is not a plural form. I think that
> Codec has ...

is the correct option

**Comment by OptimumOpium** [comment]
@jean-michel-perli-sonsin Please, approve the merge if everything is ok in your opinion

File: README.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (jean-michel-perli-sonsin) [reply README.md:45]: "Has" is used with the pronouns he, she, it and with singular nouns. So in this case, the codec has is correct.

**Review by jean-michel-perli-sonsin — CHANGES_REQUESTED** [review]

**Comment by OptimumOpium** [comment]
@jean-michel-perli-sonsin As far as I can see, all change requests were resolved. Could you please approve the merge is everything is ok?

File: README.md [hunk 9]
```diff
@@ -1,42 +1,108 @@
...
+This project is only an adapter between the Sailfish and the th2 packed into Docker Image.
+This [th2-codec-generic](https://github.com/th2-net/th2-codec-generic) project uses this image as a base.
+
 # Running
 
-To start a codec it is required to place external codec api implementation jar with dependencies to 'codec_implementation' folder.
+The codec requires an implementation of the external codec API.
+The JAR file with that implementation and all its dependencies need to be placed to the folder `home/codec_implementation`, in order to start the codec.
+
+The codec loads all JAR files from that directory and looks for all implementations of
```
Comment (jean-michel-perli-sonsin) [inline README.md:24]: The codec loads all JAR files from that directory and looks for all **the** implementations of

**Review by jean-michel-perli-sonsin — CHANGES_REQUESTED** [review]

**Review by Nikita-Smirnov-Exactpro — APPROVED** [review]

**Pull request merged**